src/test/ui/associated-type-bounds/bad-bounds-on-assoc-in-trait.stderr

```diff
@@ -25,8 +27,14 @@ LL |     type C: Clone + Iterator<Item: Send + Iterator<Item: for<'a> Lam<&'a u8
 note: required by a bound in `Iterator`
   --> $SRC_DIR/core/src/iter/traits/iterator.rs:LL:COL
    |
-LL | pub trait Iterator {
-   | ^^^^^^^^^^^^^^^^^^ required by this bound in `Iterator`
+LL | / pub trait Iterator {
+LL | |     /// The type of the elements being iterated over.
+LL | |     #[stable(feature = "rust1", since = "1.0.0")]
+LL | |     type Item;
+...  |
+LL | |     }
+LL | | }
+   | |_^ required by this bound in `Iterator`
```

Comment:
Seeing all of these I think we need to extend the metadata to have clearer wording here. We are pointing at a whole trait while saying "this bound in the trait". The _trait itself_ is "the bound". I don't know what better wording would be.